java/src/org/openqa/selenium/remote/http/jdk/JdkHttpClient.java

```diff
@@ -123,28 +125,34 @@ protected PasswordAuthentication getPasswordAuthentication() {
       builder = builder.authenticator(authenticator);
     }
 
-    Proxy proxy = config.proxy();
-    if (proxy != null) {
-      ProxySelector proxySelector =
-          new ProxySelector() {
-            @Override
-            public List<Proxy> select(URI uri) {
-              if (proxy == null) {
-                return List.of();
-              }
-              if (uri.getScheme().toLowerCase().startsWith("http")) {
-                return List.of(proxy);
-              }
+    String proxyHost = System.getProperty("http.proxyHost");
+    String proxyPort = System.getProperty("http.proxyPort");
+
+    Proxy proxy =
+        (proxyHost != null && proxyPort != null)
+            ? new Proxy(
+                Proxy.Type.HTTP, new InetSocketAddress(proxyHost, Integer.parseInt(proxyPort)))
+            : config.proxy();
```

Comment:
@pujagani I think an explicit set `config.proxy()` should beat the global system property?
And is it needed to implement this, this is just the [standard behaviour](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/net/doc-files/net-properties.html) of the HttpClient and should work out of the box?

Reply:
PS: I use `java.net.useSystemProxies=true` to pickup the system settings

Reply:
I had the same thought process before I made the code change. I tried using https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/net/doc-files/net-properties.html, and also tried digging in the HttpClient client at the part where the proxy is set, it didn't seem to use the standard system properties.